### ISSUE TITLE
Error when using ResolveField with propertyName

### ISSUE THREAD

**Issue by xballoy** [Hatko/eslint-plugin-nestjs-graphql#1]
When using `ResolveField` with the property name as recommended by Nest (see https://docs.nestjs.com/graphql/resolvers#query-type-names), the linting crashes with the following error.

```
Error: Unexpected first argument type - please contact the author of the rule
Occurred while linting /home/projects/nestjs-typescript-starter-yck1u8/src/authors.resolver.ts:28
Rule: "nestjs-graphql/matching-return-type"
    at parseDecorators (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint-plugin-nestjs-graphql/dist/matching-return-type.rule.js:172:15)
    at processNode (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint-plugin-nestjs-graphql/dist/matching-return-type.rule.js:50:48)
    at MethodDefinition[decorators.length>=1]:exit (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint-plugin-nestjs-graphql/dist/matching-return-type.rule.js:38:81)
    at ruleErrorHandler (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1091:28)
    at emit/< (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/safe-emitter.js:45:66)
    at emit (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/safe-emitter.js:45:38)
    at applySelector (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/node-event-generator.js:297:26)
    at applySelectors (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/node-event-generator.js:326:22)
    at leaveNode (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/node-event-generator.js:349:14)
    at leaveNode (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:824:23)
    at runRules/< (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1128:32)
    at runRules (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1121:15)
    at _verifyWithoutProcessors (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1370:31)
    at _verifyWithConfigArray (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1822:21)
    at verify (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:1452:65)
    at verifyAndFix (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/linter/linter.js:2083:29)
    at verifyText (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/cli-engine/cli-engine.js:254:48)
    at executeOnFiles (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/cli-engine/cli-engine.js:834:38)
    at lintFiles (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/eslint/eslint.js:551:23)
    at execute (/home/projects/nestjs-typescript-starter-yck1u8/node_modules/eslint/lib/cli.js:391:36)
```

A quick fix would be not to throw an error so the linter can work, this is currently preventing eslint to run. However it would be best if the plugin would handle this case.

Indeed, if we check at `resolved-field.decorator.d.ts` (from `@nest/graphql`) we see that there are two overrides of the method:

```
/**
 * Field resolver (method) Decorator.
 */
export declare function ResolveField(typeFunc?: ReturnTypeFunc, options?: ResolveFieldOptions): MethodDecorator;
/**
 * Property resolver (method) Decorator.
 */
export declare function ResolveField(propertyName?: string, typeFunc?: ReturnTypeFunc, options?: ResolveFieldOptions): MethodDecorator;
//# sourceMappingURL=resolve-field.decorator.d.ts.map
```

Minimal repository that reproduce the issue: https://stackblitz.com/edit/nestjs-typescript-starter-yck1u8?file=src%2Fauthors.resolver.ts

If you run `npm run lint` you'll have the error.

**Comment by Hatko** [comment]
@xballoy Thanks for the feedback - resolved in `1.1.2`

**Comment by xballoy** [comment]
Thanks! 🎉
